### ISSUE TITLE
re-add h5py support in cache_files

### ISSUE THREAD

**Issue by erwanp** [radis/radis#239]
### 🎯 Todo

h5py was replaced by pytables in c92037e  (0.9.28), for consistency with the rest of the project.

However, other spectral codes like [pySpecair](https://spectralfit.gitlab.io/specair/#) used these functions with h5py-generated files. To maintain compatbility we should keep the h5py engine as an alternative. 

### 💡 Possible implementations

Re-add: 

```

if engine=='h5py':
    # Get attributes (metadata+version)
    hf = h5py.File(file, "r")
    try:
        attrs = dict(hf.attrs)
    except OSError:
        attrs = {}
    finally:
        hf.close()

    # Raise an error if version is not found
    try:
        file_version = attrs.pop("version")
    except KeyError:
        raise DeprecatedFileError(
            "File {0} has been generated in a deprecated ".format(file)
            + "version. Delete it to regenerate it on next run"
        )

```

with an `engine='pytables'` as default

**Comment by dxb0920** [comment]
Dear sir, When I installed the pyspecair software in the linux system, according to the installation tutorial "git clone [git@gitlab.com](mailto:git@gitlab.com):spectralfit/specair.git", an error "sign_and_send_pubkey: signing failed: agent refused operation
Permission denied (public key). fatal: Could not read from remote repository.". On the gitlab.com website, I can't find any information about pyspecair. I tried to contact the administrator of the specair software, but have not received a reply yet. How should I solve this problem? Thanks.
